src/main.ts

```diff
@@ -1309,7 +1327,7 @@ function generateFromPartial(ctx: Context, fullName: string, messageDesc: Descri
         chunks.push(code`
           Object.entries(object.${fieldName}).forEach(([key, value]) => {
             if (value !== undefined) {
-              message.${fieldName}[${i}] = ${readSnippet('value')};
+              message.${fieldName}![${i}] = ${readSnippet('value')};
```

Comment:
This exclamation mark can be avoided if the code is rewritten using `reduce`. Then there is only a single assignment to `message.${fieldName}`. I did the same with `map` already for lists.

Clearly out of scope for this PR but I'll have a look.

Reply:
https://github.com/stephenh/ts-proto/pull/410

Reply:
Okay, if you rebase on latest main, you don't need this diff here anymore because there is a single assignment of `T` to `message.fiedlName: T?` now.

Reply:
Perfect, will do! Thanks!